### ISSUE TITLE
Segmentation fault in Lazy.jl

### ISSUE THREAD

**Issue by joaquimg** [JuliaLang/julia#20883]
This error has already occurred in two linux machines running julia-0.5.0 and Lazy 11.5

This only happens when I use `using Lazy` and Lazy was not installed the usual way. When it done regularly with Pkg.add("Lazy") it works just fine. However, when I download it from github and add it to som arbitrary folder (which is added to the Base.LOAD_PATH)

Any ideas?

```
julia> using Lazy
INFO: Precompiling module Lazy.

signal (11): Segmentation fault
while loading /root/sddp2/SDDPjung-distribution/JuliaPkgs/v0.5/Lazy.jl-0.11.5/Lazy/src/macros.jl, in expression starting on line 205
julia_type_to_llvm at /home/centos/buildbot/slave/package_tarball64/build/src/cgutils.cpp:318
mark_julia_const at /home/centos/buildbot/slave/package_tarball64/build/src/codegen.cpp:641
emit_expr at /home/centos/buildbot/slave/package_tarball64/build/src/codegen.cpp:3140
emit_invoke at /home/centos/buildbot/slave/package_tarball64/build/src/codegen.cpp:2669 [inlined]
emit_expr at /home/centos/buildbot/slave/package_tarball64/build/src/codegen.cpp:3162
emit_jlcall at /home/centos/buildbot/slave/package_tarball64/build/src/codegen.cpp:2578
emit_call at /home/centos/buildbot/slave/package_tarball64/build/src/codegen.cpp:2739
emit_expr at /home/centos/buildbot/slave/package_tarball64/build/src/codegen.cpp:3172
emit_function at /home/centos/buildbot/slave/package_tarball64/build/src/codegen.cpp:4691
jl_compile_linfo at /home/centos/buildbot/slave/package_tarball64/build/src/codegen.cpp:809
jl_compile_for_dispatch at /home/centos/buildbot/slave/package_tarball64/build/src/gf.c:1310
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:184 [inlined]
jl_apply_generic at /home/centos/buildbot/slave/package_tarball64/build/src/gf.c:1942
@capture at /root/sddp2/SDDPjung-distribution/JuliaPkgs/v0.5/MacroTools.jl-0.3.5/MacroTools/src/macro.jl:61
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:189 [inlined]
fl_invoke_julia_macro at /home/centos/buildbot/slave/package_tarball64/build/src/ast.c:161
apply_cl at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:1227
_applyn at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:683
fl_map1 at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:2220
apply_cl at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:1227
_applyn at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:683
fl_map1 at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:2220
apply_cl at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:1227
_applyn at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:683
fl_map1 at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:2220
apply_cl at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:1227
do_trycatch at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:899
apply_cl at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:1807
_applyn at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:683
fl_applyn at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:723
jl_call_scm_on_ast at /home/centos/buildbot/slave/package_tarball64/build/src/ast.c:778 [inlined]
jl_macroexpand at /home/centos/buildbot/slave/package_tarball64/build/src/ast.c:794
docm at ./docs/Docs.jl:605
docm at ./docs/Docs.jl:605
unknown function (ip: 0x7f653d64a564)
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:189 [inlined]
jl_apply_generic at /home/centos/buildbot/slave/package_tarball64/build/src/gf.c:1942
jl_apply_2va at /home/centos/buildbot/slave/package_tarball64/build/src/builtins.c:430
@doc at ./boot.jl:337
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:189 [inlined]
fl_invoke_julia_macro at /home/centos/buildbot/slave/package_tarball64/build/src/ast.c:161
apply_cl at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:1227
do_trycatch at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:899
apply_cl at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:1807
_applyn at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:683
fl_applyn at /home/centos/buildbot/slave/package_tarball64/build/src/flisp/flisp.c:723
jl_parse_eval_all at /home/centos/buildbot/slave/package_tarball64/build/src/ast.c:702
jl_load at /home/centos/buildbot/slave/package_tarball64/build/src/toplevel.c:596
jl_load_ at /home/centos/buildbot/slave/package_tarball64/build/src/toplevel.c:605
include_from_node1 at ./loading.jl:488
unknown function (ip: 0x7f674ad1dd0b)
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:189 [inlined]
jl_apply_generic at /home/centos/buildbot/slave/package_tarball64/build/src/gf.c:1942
do_call at /home/centos/buildbot/slave/package_tarball64/build/src/interpreter.c:66
eval at /home/centos/buildbot/slave/package_tarball64/build/src/interpreter.c:190
jl_toplevel_eval_flex at /home/centos/buildbot/slave/package_tarball64/build/src/toplevel.c:558
jl_eval_module_expr at /home/centos/buildbot/slave/package_tarball64/build/src/toplevel.c:196
jl_toplevel_eval_flex at /home/centos/buildbot/slave/package_tarball64/build/src/toplevel.c:465
jl_parse_eval_all at /home/centos/buildbot/slave/package_tarball64/build/src/ast.c:717
jl_load at /home/centos/buildbot/slave/package_tarball64/build/src/toplevel.c:596
jl_load_ at /home/centos/buildbot/slave/package_tarball64/build/src/toplevel.c:605
include_from_node1 at ./loading.jl:488
unknown function (ip: 0x7f674ad1dd0b)
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:189 [inlined]
jl_apply_generic at /home/centos/buildbot/slave/package_tarball64/build/src/gf.c:1942
do_call at /home/centos/buildbot/slave/package_tarball64/build/src/interpreter.c:66
eval at /home/centos/buildbot/slave/package_tarball64/build/src/interpreter.c:190
jl_toplevel_eval_flex at /home/centos/buildbot/slave/package_tarball64/build/src/toplevel.c:558 [inlined]
jl_toplevel_eval at /home/centos/buildbot/slave/package_tarball64/build/src/toplevel.c:580
jl_toplevel_eval_in_warn at /home/centos/buildbot/slave/package_tarball64/build/src/builtins.c:590
macro expansion; at ./none:2 [inlined]
anonymous at ./<missing> (unknown line)
unknown function (ip: 0x7f653d635681)
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:189 [inlined]
jl_toplevel_eval_flex at /home/centos/buildbot/slave/package_tarball64/build/src/toplevel.c:569 [inlined]
jl_toplevel_eval at /home/centos/buildbot/slave/package_tarball64/build/src/toplevel.c:580
jl_toplevel_eval_in_warn at /home/centos/buildbot/slave/package_tarball64/build/src/builtins.c:590
eval at ./boot.jl:234
unknown function (ip: 0x7f674ad094df)
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:189 [inlined]
jl_apply_generic at /home/centos/buildbot/slave/package_tarball64/build/src/gf.c:1942
process_options at ./client.jl:239
_start at ./client.jl:318
unknown function (ip: 0x7f674ad432e8)
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:189 [inlined]
jl_apply_generic at /home/centos/buildbot/slave/package_tarball64/build/src/gf.c:1942
jl_apply at /home/centos/buildbot/slave/package_tarball64/build/ui/../src/julia.h:1392 [inlined]
true_main at /home/centos/buildbot/slave/package_tarball64/build/ui/repl.c:112
main at /home/centos/buildbot/slave/package_tarball64/build/ui/repl.c:232
__libc_start_main at /lib64/libc.so.6 (unknown line)
unknown function (ip: 0x4013fc)
Allocations: 1236013 (Pool: 1235138; Big: 875); GC: 0
ERROR: Failed to precompile Lazy to /root/sddp2/SDDPjung-distribution/JuliaPkgs/lib/v0.5/Lazy.ji.
 in compilecache(::String) at ./loading.jl:593
 in require(::Symbol) at ./loading.jl:422


```

**Comment by malmaud** [comment]
This should probably be filed on https://github.com/MikeInnes/Lazy.jl/issues.

**Comment by joaquimg** [comment]
My bad, typed the wrong page. Anyway it was just a problem in MacroTools 0.3.5 that was resolved in 0.3.6
